python/mlx/nn/layers/convolution.py

```diff
@@ -45,25 +46,29 @@ def __init__(
         self.weight = mx.random.uniform(
             low=-scale,
             high=scale,
-            shape=(out_channels, kernel_size, in_channels),
+            shape=(out_channels, kernel_size, int(in_channels / groups)),
```

Comment:
I think it would be good to add a check above if `in_channels` is not divisible by groups and raise a `ValueError`.

Reply:
Sure thing, I added it.

Reply:
It is an absolute nitpick but perhaps also use `in_channels // groups` instead of `int(in_channels / groups)` here?

Reply:
Yep, no problem, I updated it.